### PR TITLE
STOP LIMIT FInal PR

### PR DESCRIPTION
**Pull request by HarryTgerman**

### CODE REVIEW

File: packages/limit-orders-lib/src/constants/index.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
 export const ETH_ADDRESS = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
 
 export const SLIPPAGE_BPS = 40; // 0.4%
-export const TWO_BPS_GELATO_FEE = 2; // 0.02%
+export const STOP_LIMIT_SLIPPAGE_BPS = 500; // 5%
+export const TWO_BPS_GELATO_FEE = 10; // 0.1%
```
Comment (gitpusha) [inline packages/limit-orders-lib/src/constants/index.ts:5]: rename to simply BPS_GELATO_FEE

**Comment by gitpusha** [comment]
Check this issue for ExpiryDates: https://github.com/gelatodigital/swap-orders-lib/issues/17

File: packages/limit-orders-lib/src/utils/queries/index.ts [hunk 2]
```diff
@@ -171,16 +179,33 @@ export const queryCancelledOrders = async (
...
   }
 };
 
-const _getUniqueOrdersWithHandler = (allOrders: Order[]): Order[] =>
-  [...new Map(allOrders.map((order) => [order.id, order])).values()]
+const checkExpiration = (allOrders: Order[]): Order[] =>
+  allOrders.map((order: Order) => {
+    order.isExpired =
+      Date.now() > (parseInt(order.createdAt) + MAX_LIFETIME) * 1000;
+    return { ...order };
```
Comment (gitpusha) [inline packages/limit-orders-lib/src/utils/queries/index.ts:194]: why not `return order` ?

**Comment by gitpusha** [comment]
Improvement: 

- getHandler directly from subgraph everywhere in queries: packages/limit-orders-lib/src/utils/queries/constants.ts

- handler is now string | null on every Order 


- replace `_getUniqueOrdersWithHandler`

File: packages/limit-orders-lib/src/utils/queries/stoplimit.ts [hunk 3]
```diff
@@ -64,19 +65,19 @@ export const queryStopLimitExecutedOrders = async (
...
             owner: owner.toLowerCase(),
             module:
-              GELATO_STOPLOSS_ORDERS_MODULE_ADDRESS[chainId].toLowerCase(),
+              GELATO_STOP_LIMIT_ORDERS_MODULE_ADDRESS[chainId].toLowerCase(),
           }
         )
       : { orders: [] };
 
-    const orders = dataStopLimitSubgraph.orders;
+    const orders = [...dataStopLimitSubgraph.orders];
```
Comment (gitpusha) [inline packages/limit-orders-lib/src/utils/queries/stoplimit.ts:80]: just use array

File: packages/limit-orders-lib/src/utils/queries/stoplimit.ts [hunk 4]
```diff
@@ -37,19 +38,19 @@ export const queryOpenStopLimitOrders = async (
...
```
Comment (gitpusha) [inline packages/limit-orders-lib/src/utils/queries/stoplimit.ts:53]: just use array

File: packages/limit-orders-lib/src/utils/queries/stoplimit.ts [hunk 5]
```diff
@@ -140,20 +141,22 @@ export const queryPastOrders = async (
   }
 };
 
+const checkExpiration = (allOrders: StopLimitOrder[]): StopLimitOrder[] =>
```
Comment (gitpusha) [inline packages/limit-orders-lib/src/utils/queries/stoplimit.ts:144]: change to addExpiration

File: packages/limit-orders-lib/src/utils/queries/stoplimit.ts [hunk 6]
```diff
@@ -140,20 +141,22 @@ export const queryPastOrders = async (
...
 export const _getUniqueOrdersWithHandler = (
   allOrders: StopLimitOrder[]
 ): StopLimitOrder[] =>
-  [...new Map(allOrders.map((order) => [order.id, order])).values()]
+  [
+    ...new Map(
+      checkExpiration(allOrders)
+        // filter out wrong module
+        .map((order) => [order.id, order])
+    ).values(),
```
Comment (gitpusha) [inline packages/limit-orders-lib/src/utils/queries/stoplimit.ts:159]: replace with _getUniqueOrdersWithExpiry

First get array of unique orders

Then addExpiry to them and return

**Review by 0mkara — APPROVED** [review]

**Pull request merged**